src/strategies/keystore.rs

```diff
@@ -211,4 +212,27 @@ mod tests {
         keystore.release_key(addr2).await.unwrap();
         keystore.release_key(acquired_addr).await.unwrap();
     }
+
+    #[tokio::test]
+    async fn test_get_address_randomness() {
+        let mut keystore = KeyStore::new();
+        keystore.add_key("address1".to_string(), "private_key1".to_string()).await;
+        keystore.add_key("address2".to_string(), "private_key2".to_string()).await;
+        keystore.add_key("address3".to_string(), "private_key3".to_string()).await;
+
+        let mut counts = HashMap::new();
+        let iterations = 1000;
+
+        for _ in 0..iterations {
+            if let Some(address) = keystore.get_address() {
+                *counts.entry(address).or_insert(0) += 1;
+            }
+        }
+
+        // Check that each address was selected roughly equally
+        for count in counts.values() {
+            let proportion = *count as f64 / iterations as f64;
+            assert!((0.3..0.4).contains(&proportion), "Proportion out of expected range");
```

Comment:
We'd expect them to converge to .33 right? Let's add more buffer on the lower end since .3 is a bit close.

Reply:
sounds good. now doing 100x more iterations with stricter bound